### DISCUSSION TITLE
Security concern: Unauthorized access to private channels by setting `private: false`?

### DISCUSSION THREAD

**Discussion by Ynng** [supabase/supabase#29334] [Questions] [answered]
I'm developing an app that syncs private data in real-time for each user.
The implementation uses one channel per user, with the topic name being the user's id with a prefix.
I've enabled RLS on `realtime.messages` and set up policies to restrict channel access based on `realtime.topic()` and `auth.uid()`

The policy works as expected for supabase clients with `private: true` in their client channel config.

However, I found that I can connect to any channel and listen as another user, as long as I declare "private: false" when subscribing to the channel.

Is there any way to prevent this? How can I ensure that only authorized users can access their respective private channels?"

**Comment by GaryAustin1** [comment, marked as answer]
A non-private channel should not be able to see info on a private channel.  But if you use the same channel name there will be two different channels with the same name.  One will require private:true for both sender and receiver, the other would require private:false (or no setting) for both ends and would be a public channel.
